src_c/rect.c

```diff
@@ -342,19 +342,34 @@ pgRect_Normalize(SDL_Rect *rect)
 static int
 _pg_do_rects_intersect(SDL_Rect *A, SDL_Rect *B)
```

Comment:
Did you do a performance test without the changes to this function?

The changes you've made below seems like details that the compiler would deal with to make an efficient program, so I just wonder if the changes to this function actually make any difference.

Reply:
Right now i'm not at home and i'm afraid i'm not able to make super precise measurements, but i measured a performance difference with the previous iteration (all documented) with two more intersection helpers for normalized and not normalized rects. That added a lot of code duplication and in the end i thought to merge the changes into the base function and just use one. I remember i gave this helper a new run of tests and it seemed a bit better than before, but again i think i'll retest this in the coming days.

Reply:
Fine, I'll test it myself :)

Reverting the changes to `_pg_do_rects_intersect` did not yield a statistically significant performance difference.

So I would like these changes to be reverted to make this PR simpler.